src/groupeddataframe/grouping.jl

```diff
@@ -315,26 +315,30 @@ function Base.map(f::Any, gd::GroupedDataFrame)
 end
 
 """
-    combine(gd::GroupedDataFrame, cols => f...)
+    combine(gd::GroupedDataFrame, (cols => f)...)
+    combine(gd::GroupedDataFrame, [cols1 => f1, cols2 => f2]...)
     combine(gd::GroupedDataFrame; (colname = cols => f)...)
     combine(gd::GroupedDataFrame, f)
     combine(f, gd::GroupedDataFrame)
 
 Transform a [`GroupedDataFrame`](@ref) into a `DataFrame`.
 
-If the last argument(s) consist(s) in one or more `cols => f` pair(s), or if
-`colname = cols => f` keyword arguments are provided, `cols` must be
-a column name or index, or a vector or tuple thereof, and `f` must be a callable.
-A pair or a (named) tuple of pairs can also be provided as the first or last argument.
-If `cols` is a single column index, `f` is called with a `SubArray` view into that
-column for each group; else, `f` is called with a named tuple holding `SubArray`
-views into these columns.
+The last argument(s) in `combine` can be either:
 
-If the last argument is a callable `f`, it is passed a [`SubDataFrame`](@ref) view for each group,
-and the returned `DataFrame` then consists of the returned rows plus the grouping columns.
-Note that this second form is much slower than the first one due to type instability.
-A method is defined with `f` as the first argument, so do-block
-notation can be used.
+* One or several `cols => f` pairs, or vectors or tuples of such pairs (mixing is allowed). `cols`
+  must be a column name or index in `gd`, or a vector or tuple thereof. `f` must be callable.
+  If `cols` is a single column index, `f` is called with a `SubArray` view into that
+  column for each group; else, `f` is called with a named tuple holding `SubArray`
+  views into these columns.
+* A named tuple of `colname = cols => f` pairs or keyword arguments of such pairs,
+  where `colname` indicates the name of the column to be created in the new `DataFrame`.
+  Pairs must obey the same rules as above.
+* A callable `f` taking a `SubDataFrame` view for each group. The returned `DataFrame` 
+  then consists of the returned rows plus the grouping columns.
+  Note that this form is much slower than the others due to type instability.
+
+A method is defined with `f` as the first argument, so do-block notation can be used.
+In that case `f` can also be a named tuple of pairs.
 
 `f` can return a single value, a row or multiple rows. The type of the returned value
```

Comment:
The restriction is if we write `colname = cols=f` it must return a column. Right?

Reply:
For the `Pairs`, including named tuple stuff, you can only return a scalar or a vector. The vectors get stacked. All pairs must return something of the same length. 

Tuples of vectors do not get splatted. For instance `combine(gd, :b => x -> (x, x .+ 1))` does not create two columns.

Reply:
Thank you. This is my understanding, but I could not find where we specify this in the docstring.

Reply:
That's mentioned below:
> As a special case, if a tuple or vector of pairs is passed as the first argument, each function
is required to return a single value or vector, which will produce each a separate column.

It's really hard to find a good way to organize all that stuff given how many possible combinations there are.